### PR TITLE
Make "decimal" functions more consistent

### PR DESCRIPTION
**Pull request by aaronfranke**
Previously, "decimals()" in GDScript and "Decimals()" in C# had different behavior. I have renamed the method to "step_decimals()". It was already called that internally. C# is now consistent with core.

### CODE REVIEW

**Comment by akien-mga** [comment]
I don't think `decimals` is the right name for a function that gives the fractional part of a number - the plural doesn't make sense, it should just be `decimal`.

That said, we're not keen on breaking compatibility unless there is a strong incentive, so I'm not convinced by these changes.

**Comment by CptPotato** [comment]
I find the name `decimal` unfitting for a function that gives you the fractional part of a (decimal) number.
How about naming it `frac()` or `fract()` as used in shader languages?

**Comment by groud** [comment]
I don't even know why we need this function, which can easily be solved by `fmod(myvar, 1.0)`

**Comment by CptPotato** [comment]
You're right, though modulo is a bit slower.

**Comment by aaronfranke** [comment]
Then should I remove the new "decimals()" functionality and instead leave it pointing to "step_decimals" like it was before? That would make it not break compatibility.

**Comment by akien-mga** [comment]
IMO it's not worth renaming `decimals` to `step_decimals` at this stage, but we can add it in #16863 to the list of methods that we want to rename next time we break compatibility.

**Comment by avencherus** [comment]
I may be missing something, but the way I was reading it was that it returns and integer and is _trying_ to count decimal places.  Seems a bit tricky at the far ends of a float/double if you're doing operations on them.

I usually see this type of utility function called `decimal_places()`

https://github.com/godotengine/godot/blob/4f5a7ebaecfcf00cf1e5c4af4b20034f0dcecd29/core/math/math_funcs.cpp#L59-L81

@aaronfranke If you don't mind answering, I was curious about how this works.  X)

`return BitConverter.GetBytes(decimal.GetBits(s)[3])[2];`

**Comment by akien-mga** [comment]
No longer breaks compat.

If we're deprecating `decimals` to replace it by `step_decimals`, I'd add something like this to `GDScriptFunctions::call` and `Expression::exec_func`:
```cpp
ERR_EXPLAIN("GDScript method 'decimals' is deprecated and replaced by 'step_decimals', please update your code accordingly.");
WARN_DEPRECATED
```
(and then still fallback to `step_decimals` handling so that it works even if deprecated).

About the addition of `DecimalsCount` in C#, that's up to @neikeq. I've already mentioned how I feel about adding C#-only features when there's no reason why C# users would need them more than GDScript users. And the whole point of using C# in the first place is to have access to the huge ecosystem of C# libraries, many of which probably cover such method already so that we don't have to reimplement the wheel in core (beyond the compat layer with GDScript).

**Comment by akien-mga** [comment]
Since 74eefda56 is independent of the renaming proposal, I'd suggest to put it in its own PR so that I can merge it as a bugfix.

**Comment by aaronfranke** [comment]
Here it is then: https://github.com/godotengine/godot/pull/22251

**Comment by aaronfranke** [comment]
@akien-mga I've added the deprecation warnings as you suggested.

And to clarify, as I mentioned in https://github.com/godotengine/godot/pull/22251, the only reason `DecimalCount` exists is because the C# code had this functionality already, but under the "Decimals" name. It can be removed if requested (and/or in the future), and I don't need it personally, I'm just trying to keep the existing functionality.

File: doc/classes/@GDScript.xml [hunk 1]
```diff
@@ -264,10 +264,14 @@
 			<argument index="0" name="step" type="float">
 			</argument>
 			<description>
-				Returns the position of the first non-zero digit, after the decimal point.
+				Deprecated alias for "step_decimals". Returns the position of the first non-zero digit, after the decimal point.
```
Comment (akien-mga) [inline doc/classes/@GDScript.xml]: You can use `[method step_decimals]` for a direct link to that method. I'd also remove the rest of the documentation for this one, only mention that it's deprecated by `step_decimals`.

**Review by akien-mga — APPROVED** [review]
I still don't have a strong opinion on this matter but it's being deprecated the right way, so that's fine with me to merge. If anyone is against it, speak now or never :)

**Comment by aaronfranke** [comment]
So, just to clarify the justification for this PR:

`decimals` was a confusing name for this method, to the point that I was confused by it, and the previous C# implementation was wrong. So it's been renamed, and C# is now consistent with Core.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**